### PR TITLE
Update user-guide to point to generic docs

### LINKED ISSUE #215: Point to generic docs
Now that we have the generic Application Runtimes Operator, we should try to reference the [docs](https://github.com/application-runtimes/operator/tree/master/doc) from that repo as much as possible to avoid duplicating information.  It should be possible to have a link to the places they are the same and just talk about the overlay - such as which `Kind` to use and extra features.

### PR DESCRIPTION
**Pull request by edavidj**
**What this PR does / why we need it?**:

- Reduce duplication of information across the different operators by pointing to the application-stacks/operator repo where applicable.
- Currently approaches the task with a list of labelled links pointing to documentation that is housed in app stacks user-guide. If there's another desired approach, I can change it 👍 

**Which issue(s) this PR fixes**:

Fixes #215

### CODE REVIEW

File: doc/user-guide.md [hunk 1]
```diff
@@ -124,402 +124,30 @@ oc get appsodyapplication my-appsody-app
 oc get app my-appsody-app
 ```
 
-### Image Streams
+### Common Component Documentation
+
+Appsody Application Operator is based on the generic [Application Stacks
+Operator](https://github.com/application-stacks/operator). To see more
+information on the usage of common functionality, see Application Stacks documentation:
```
Comment (arthurdm) [inline doc/user-guide.md]: I like this approach.  The only thing I would add is something like "In the samples from the links below, just replace `Kind: RuntimeComponent` with `Kind: AppsodyApplication`"

**Comment by arthurdm** [comment]
thanks @edavidj - looks great.  I added an inline suggestion, and also noticed this PR kept the sections `### Kubernetes Application Navigator (kAppNav) support` and `### Certificate Manager support` - I think those could be removed and referenced from the list of links?

File: doc/user-guide.md (hunk 1 above)
Comment (edavidj) [reply doc/user-guide.md]: Sure thing, I'll add the message right before the list of links

**Comment by arthurdm** [comment]
thanks for fixing it up!   I think there's some lingering sub-section from the Cert Manager: the cert manager is at level `###` and there's some sub-sections are level `####` that show the Route, etc, which are already in the generic doc and could be removed.  Basically, can remove everything up to just before the Troubleshooting link.

**Pull request merged**